ewatercycle/models/lisflood.py

```diff
@@ -241,6 +241,43 @@ def get_value_as_xarray(self, name: str) -> xr.DataArray:
 
         return da
 
+    def coords_to_indices(self, name: str, lat: Iterable[float], lon: Iterable[float]) -> Iterable[int]:
+        """Convert lat, lon coordinates into model indices."""
+        grid_id = self.bmi.get_var_grid(name)
+        shape = self.bmi.get_grid_shape(grid_id) #(len(y), len(x))
+        x_model = self.bmi.get_grid_x(grid_id)
+        y_model = self.bmi.get_grid_y(grid_id)
+        indices = []
+        coord_converted = []
+        # in lisflood, x corresponds to lon, and y to lat.
+        # this might not be the case for other models!
+        for x, y in zip(lon, lat):
+            idx = np.abs(x_model - x).argmin()
+            idy = np.abs(y_model - y).argmin()
```

Comment:
The current function [lat_lon_to_closest_variable_indices](https://github.com/eWaterCycle/ewatercycle/blob/a7c8d3a2e9ff4ccfaa7f5cd1e503d3e8818c9ce5/ewatercycle/util.py#L43) in util.py uses absolute distance. Here, it is now changed to euclidean distance. BTW, both types of calculations are only valid for small distances because coordinates are lat/lon. Also, I added a threshold.